### PR TITLE
statfs support subdir based on dir quota

### PR DESCRIPTION
**Pull request by Hexilee**

### CODE REVIEW

File: pkg/meta/base.go [hunk 1]
```diff
@@ -819,8 +819,70 @@ func (m *baseMeta) cleanupSlices() {
...
+		if q == nil {
+			continue
+		}
+		if usage == nil {
+			usage = q
+		}
+		if q.MaxSpace > 0 {
+			ls := q.MaxSpace - q.UsedSpace
+			if ls < 0 {
+				logger.Errorf("invalid quota for inode %d: %+v", root, *q)
```
Comment (davies) [inline pkg/meta/base.go:853]: it's OK,  left space is 0

File: pkg/meta/base.go [hunk 2]
```diff
@@ -819,8 +819,70 @@ func (m *baseMeta) cleanupSlices() {
...
+func (m *baseMeta) StatFS(ctx Context, totalspace, availspace, iused, iavail *uint64, subdir bool) syscall.Errno {
 	defer m.timeit(time.Now())
+	if st := m.statRootFs(ctx, totalspace, availspace, iused, iavail); st != 0 {
+		return st
+	}
+	if m.root == RootInode || !subdir {
+		return 0
+	}
+	var usage *Quota
+	var leftSpace, leftInodes int64 = -1, -1
```
Comment (davies) [inline pkg/meta/base.go:831]: use availspace as initial value

Comment (Hexilee) [reply pkg/meta/base.go:831]: We need decide whether override `totalspace` or `iused` based on `leftSpace` or `leftInodes` is less than 0

File: pkg/meta/base.go (hunk 1 above)
Comment (Hexilee) [reply pkg/meta/base.go:853]: Ok

Comment (Hexilee) [reply pkg/meta/base.go]: Done

File: pkg/meta/base.go (hunk 2 above)
Comment (Hexilee) [reply pkg/meta/base.go]: My fault, fixed

**Pull request merged**